cmd/cpe2cve/config.go

```diff
@@ -0,0 +1,300 @@
+
+	// optimizations
+	flag.IntVar(&c.nProcessors, "nproc", 1, "number of concurrent goroutines that perform CVE lookup")
+	flag.BoolVar(&c.indexedDict, "idxd", false, "build and use an index for CVE dictionary: increases the processing speed, but might miss some matches")
+	flag.Int64Var(&c.cacheSize, "cache_size", 0, "limit the cache size to this amount in bytes; 0 removes the limit, -1 disables caching")
+	flag.BoolVar(&c.requireVersion, "require_version", false, "ignore matches of CPEs with version ANY")
+
+	// profiling
+	flag.StringVar(&c.cpuProfile, "cpuprofile", "", "file to store CPU profile data to; empty value disables CPU profiling")
+	flag.StringVar(&c.memProfile, "memprofile", "", "file to store memory profile data to; empty value disables memory profiling")
+
+	// feeds
+	flag.Var(&c.overrides, "r", "overRide: path to override feed, can be specified multiple times")
+}
+
+func (c *config) addFeedsFromArgs(args ...string) error {
+	if c.feeds == nil {
+		c.feeds = make(map[string][]string)
+	}
+	if len(args) > 0 && c.providerAt != 0 && c.provider == "" {
+		return fmt.Errorf("in order to output a provider, you need to specify a valid provider name using the -provider flag")
+	}
+	for _, file := range args {
+		c.feeds[c.provider] = append(c.feeds[c.provider], file)
+	}
+	return nil
+}
+
+func (c *config) validate() error {
+	if len(c.feeds) == 0 {
+		return fmt.Errorf("feed files weren't provided")
+	}
+	if c.cpesAt <= 0 {
+		return fmt.Errorf("-cpe flag wasn't provided")
+	}
+	if c.cvesAt <= 0 {
+		return fmt.Errorf("-cve flag wasn't provided")
+	}
+	if c.matchesAt < 0 {
+		return fmt.Errorf("-matches value is invalid %d", c.matchesAt)
+	}
+	if c.cwesAt < 0 {
+		return fmt.Errorf("-cwe value is invalid %d", c.cwesAt)
+	}
+	if c.cvss2at < 0 {
+		return fmt.Errorf("-cvss2 value is invalid %d", c.cvss2at)
+	}
+	if c.cvss3at < 0 {
+		return fmt.Errorf("-cvss2 value is invalid %d", c.cvss3at)
+	}
+	if c.cvssAt < 0 {
+		return fmt.Errorf("-cvss value is invalid %d", c.cvssAt)
+	}
+	return nil
+}
+
+type configFile struct {
+	CPEsAt             int
+	CVEsAt             int
+	MatchesAt          int
+	CWEsAt             int
+	CVSS2At            int
+	CVSS3At            int
+	CVSSAt             int
+	EraseFields        []int
+	ProviderAt         int
+	InFieldSeparator   string
+	InRecordSeparator  string
+	OutFieldSeparator  string
+	OutRecordSeparator string
+	NumProcessors      int
+	IndexDict          bool
+	CacheSize          int64
+	RequireVersion     bool
+	CPUProfile         string
+	MemoryProfile      string
+	FeedOverrides      []string
+	Feeds              map[string]string
+}
+
+func readConfigFile(file string) (*configFile, error) {
+	f, err := os.Open(file)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	switch ext := path.Ext(file); ext {
+	case ".json":
+		return readJSONConfig(f)
+	case ".toml":
+		return readTOMLConfig(f)
+	default:
```

Comment:
Have you considered making either JSON or TOML default and only check for extension once? To be clear, this approach also has merits, no strong objection here.

Reply:
I want to leave it like that to be explicit. Which one would you make explicit?